packages/@aws-cdk/aws-iam/test/integ.imported-role.ts

```diff
@@ -0,0 +1,32 @@
+import { App, Stack } from '@aws-cdk/core';
+import {
+  IAM_IMPORTED_ROLE_STACK_SAFE_DEFAULT_POLICY_NAME,
+} from '@aws-cdk/cx-api';
+import * as integ from '@aws-cdk/integ-tests';
+import { PolicyStatement, Role, ServicePrincipal } from '../lib';
+
+const app = new App({ context: { [IAM_IMPORTED_ROLE_STACK_SAFE_DEFAULT_POLICY_NAME]: true } });
+
+const roleStack = new Stack(app, 'integ-iam-imported-role-role-stack');
+
+const role = new Role(roleStack, 'TestRole', {
+  assumedBy: new ServicePrincipal('sqs.amazonaws.com'),
+});
+
+const firstStack = new Stack(app, 'integ-iam-imported-role-1');
+const roleInFirstStack = Role.fromRoleName(firstStack, 'Role', role.roleName);
+roleInFirstStack.addToPolicy(new PolicyStatement({ resources: ['firstQueue'], actions: ['sqs:SendMessage'] }));
+
+const secondStack = new Stack(app, 'integ-iam-imported-role-2');
+secondStack.addDependency(firstStack, "So that this stack can be tested after both are deployed.")
+
+const roleInSecondStack = Role.fromRoleName(secondStack, 'Role', role.roleName);
+roleInSecondStack.addToPolicy(new PolicyStatement({ resources: ['secondQueue'], actions: ['sqs:SendMessage'] }));
+
+const test = new integ.IntegTest(app, 'ImportedRoleTest', {
+  testCases: [secondStack],
+});
+
+test.assertions.awsApiCall('IAM', 'listRolePolicies', { RoleName: role.roleName }).assertAtPath('PolicyNames', integ.ExpectedResult.arrayWith([integ.Match.stringLikeRegexp('.*'), integ.Match.stringLikeRegexp('.*')]));
```

Comment:
Nice! But let's make the string match a bit more specific.

Reply:
Sure - what do you have in mind? As mentioned, I can't run the integration tests myself, which means that I don't know the exact outputs that will be created in this case.